src/test/scala/uk/gov/hmrc/test/ui/pages/section6/DepartureTransportPage.scala

```diff
@@ -19,54 +19,74 @@ package uk.gov.hmrc.test.ui.pages.section6
 import uk.gov.hmrc.test.ui.pages.base.Constants.{Clearance, Common}
 import uk.gov.hmrc.test.ui.pages.base.TariffLinks.{departureTransport, departureTransportCL}
 import uk.gov.hmrc.test.ui.pages.base.{BasePage, Details}
-import uk.gov.hmrc.test.ui.pages.section6.DetailKeys.{DepartureTransport, InlandModeOfTransport, TransportLeavingBorder}
+import uk.gov.hmrc.test.ui.pages.section1.DeclarationChoicePage.isClearance
+import uk.gov.hmrc.test.ui.pages.section6.DetailKeys.{
+  DepartureTransport,
+  InlandModeOfTransport,
+  InlandOrBorder,
+  TransportLeavingBorder
+}
 import uk.gov.hmrc.test.ui.pages.section6.InlandOrBorderPage.isBorderLocation
 
 object DepartureTransportPage extends BasePage {
 
   def backButtonHref: String =
-    maybeDetail(InlandModeOfTransport).fold(InlandModeOfTransportPage.backButtonHref)( _=> InlandModeOfTransportPage.path)
+    if (maybeDetail(InlandOrBorder).nonEmpty) {
+      detail(InlandOrBorder) match {
+        case "Border location"             => InlandOrBorderPage.path
+        case "Customs controlled location" => InlandModeOfTransportPage.path
+      }
+    } else if (isClearance) SupervisingCustomsOfficePage.path
+    else {
+      maybeDetail(InlandModeOfTransport).fold(InlandModeOfTransportPage.backButtonHref)(_ =>
+        InlandModeOfTransportPage.path
+      )
+    }
```

Comment:
I would always place the "journey" test as the 1st condition. I mean...
```
if (isClearance) SupervisingCustomsOfficePage.path
else if (maybeDetail(InlandOrBorder).nonEmpty) detail(InlandOrBorder) match {
  case "Border location"             => InlandOrBorderPage.path
  case "Customs controlled location" => InlandModeOfTransportPage.path
}
else maybeDetail(InlandModeOfTransport)
  .fold(InlandModeOfTransportPage.backButtonHref)(_ => InlandModeOfTransportPage.path)
```

Reply:
Done